### PR TITLE
[FIX] web: traceback generated by hotkeys

### PR DESCRIPTION
**Pull request by FrancoisGe**
This commit fixes the erroneous tracebacks generated by keyup events
in the hotkey overlay. (alt + key)

TASK-2591069 - BUG 3

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by brboi — APPROVED** [review]

File: addons/web/static/tests/core/hotkeys/hotkey_service_tests.js [hunk 1]
```diff
@@ -181,6 +194,44 @@ QUnit.test("non-MacOS usability", async (assert) => {
     removeHotkey();
 });
 
+QUnit.test("the overvay of hotkeys is correctly displayed", async (assert) => {
```
Comment (brboi) [inline addons/web/static/tests/core/hotkeys/hotkey_service_tests.js:197]: overlay :-)

Comment (brboi) [inline addons/web/static/tests/core/hotkeys/hotkey_service_tests.js:197]: overlay :-)

File: addons/web/static/tests/core/hotkeys/hotkey_service_tests.js [hunk 2]
```diff
@@ -181,6 +194,44 @@ QUnit.test("non-MacOS usability", async (assert) => {
...
+    assert.expect(7);
+
+    class MyComponent extends Component {
+        onClickB() {
+            assert.step("click b");
+        }
+        onClickC() {
+            assert.step("click c");
+        }
```
Comment (brboi) [inline addons/web/static/tests/core/hotkeys/hotkey_service_tests.js:206]: FYI, you could also do that
```suggestion
onClick(ev) {
    assert.step(`click ${ev.target.dataset.hotkey}`);
}
```

File: addons/web/static/tests/core/hotkeys/hotkey_service_tests.js [hunk 3]
```diff
@@ -15,6 +16,18 @@ const serviceRegistry = registry.category("services");
 let env;
 let target;
 
+function displayHotkeysOverlay() {
+    if (isMacOS()) {
+        window.dispatchEvent(new KeyboardEvent("keydown", { key: "alt", ctrlKey: true }));
+    } else {
+        window.dispatchEvent(new KeyboardEvent("keydown", { key: "alt", altKey: true }));
+    }
+}
```
Comment (brboi) [inline addons/web/static/tests/core/hotkeys/hotkey_service_tests.js:25]: No need to make a "global" function as you use it only in one test.
```suggestion
```
See below comment.

File: addons/web/static/tests/core/hotkeys/hotkey_service_tests.js [hunk 4]
```diff
@@ -15,6 +16,18 @@ const serviceRegistry = registry.category("services");
...
+
+function getOverlayHotkeys(comp) {
+    return [...comp.el.querySelectorAll(".o_web_hotkey_overlay")].map((el) => el.innerText);
+}
```
Comment (brboi) [inline addons/web/static/tests/core/hotkeys/hotkey_service_tests.js:29]: No need to make a "global" function as you use it only in one test.
```suggestion
```
See below comment.

File: addons/web/static/tests/core/hotkeys/hotkey_service_tests.js [hunk 5]
```diff
@@ -181,6 +194,44 @@ QUnit.test("non-MacOS usability", async (assert) => {
...
+        }
+    }
+    MyComponent.template = xml`
+        <div>
+        <button t-on-click="onClickB" data-hotkey="b"/>
+        <button t-on-click="onClickC" data-hotkey="c"/>
+        </div>
+    `;
+    const comp = await mount(MyComponent, { env, target });
+
```
Comment (brboi) [inline addons/web/static/tests/core/hotkeys/hotkey_service_tests.js:215]:
```suggestion
const getOverlays = () => [...comp.el.querySelectorAll(".o_web_hotkey_overlay")].map((el) => el.innerText);
```

File: addons/web/static/tests/helpers/utils.js [hunk 6]
```diff
@@ -199,6 +199,7 @@ export function triggerHotkey(hotkey, addOverlayModParts = false, eventAttrs = {
     }
 
     window.dispatchEvent(new KeyboardEvent("keydown", eventAttrs));
+    window.dispatchEvent(new KeyboardEvent("keyup", eventAttrs));
```
Comment (brboi) [inline addons/web/static/tests/helpers/utils.js:202]: Oh I see :-)
Love it. This is symmetrical !
No other test needs to get adapted with this ?

File: addons/web/static/tests/core/hotkeys/hotkey_service_tests.js [hunk 7]
```diff
@@ -181,6 +194,44 @@ QUnit.test("non-MacOS usability", async (assert) => {
     removeHotkey();
 });
 
+QUnit.test("the overvay of hotkeys is correctly displayed", async (assert) => {
+    assert.expect(7);
+
```
Comment (brboi) [inline addons/web/static/tests/core/hotkeys/hotkey_service_tests.js:199]:
```suggestion
const displayHotkeysOverlay = () => {
    const mac = isMacOS();
    window.dispatchEvent(new KeyboardEvent("keydown", { altKey: !mac, ctrlKey: mac }));
}
```

**Comment by aab-odoo** [comment]
robodoo rebase-ff r+

**Pull request closed without merging**